gems/smithy/lib/smithy/templates/client/protocol_spec.erb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+require_relative '../spec_helper'
+<% additional_requires.each do |r| -%>
+require '<%= r %>'
+<% end -%>
+
+
+module <%= module_name %>
+  # TODO: Can be replaced by stub_responses config once implemented
+  class StubSend < Smithy::Client::Plugin
+    option(:stub_response)
+    handle(step: :send) do |context|
+      if (stub_response = context.config.stub_response)
+        resp = context.response
+        resp.signal_headers(stub_response[:status_code], stub_response.fetch(:headers, {}))
+        resp.signal_data(stub_response[:body]) if stub_response[:body]
+        resp.signal_done
+      end
+      Smithy::Client::Output.new(context: context)
+    end
+  end
+
+  describe Client do
+    before(:all) { Client.add_plugin(StubSend) }
+    after(:all) { Client.remove_plugin(StubSend) }
+
+    let(:client_options) do
+      {
+        # stub_responses: true,
+        # validate_input: false,
+        endpoint: 'http://127.0.0.1',
+        # retry_strategy: # disable?
+      }
+    end
+
+    let(:client) { Client.new(client_options) }
+
+<% all_operation_tests.each do |operation_tests| %>
+    describe '#<%= operation_tests.name %>' do
+<% unless operation_tests.request_tests.empty? -%>
+      describe 'requests' do
+<% operation_tests.request_tests.each do |test| %>
+<% test.comments.each do |line| -%>
+        # <%= line %>
+<% end -%>
+<% if test.skip? -%>
+        it '<%= test.id %>', skip: '<%= test.skip_reason %>' do
+<% else -%>
+        it '<%= test.id %>' do
+<% end -%>
+<% if test['host'] -%>
+          client = Client.new(client_options.merge(endpoint: '<%= test.test_case.fetch('authScheme', 'http') %>://<%= test['host'] %>'))
```

Comment:
I think you can just merge client options because client has not been referenced or loaded yet. I would also consider putting endpoint in a method for the test or test case.

Reply:
I could do it only with `client_options.merge!` and to me it was harder to read that way (requires more knowledge of how the lets work/are setup.  harder to jsut read the code as is).   If you prefer that style though, I'll change it. 

good idea on the endpoint method.

Reply:
You can't do client_options = client_options.merge()?